server/plugin/oAuth.go

```diff
@@ -173,6 +223,23 @@ func (p *Plugin) GenerateOAuthToken(code, state string) error {
 	return nil
 }
 
+// isAccessTokenExpired checks if a user's access token is expired
+func (p *Plugin) isAccessTokenExpired(mattermostUserID string) bool {
+	user, err := p.Store.LoadUser(mattermostUserID)
+	if err != nil {
+		p.API.LogError(constants.ErrorLoadingUserData, "Error", err.Error())
+		return false
+	}
+
+	// TODO: use middleware for all such places to check if user's oAuth is completed
+	// Consider some buffer for comparing expiry time
+	if user.AccessToken != "" && user.ExpiresAt < time.Now().UTC().Add(-(time.Minute*constants.TokenExpiryTimeBufferInMinutes)).Unix() {
```

Comment:
Use the `time.Until` function here.

Reply:
here
`user.ExpiresAt` has type `int64` as it stores `Unix timestamp` and so, we are comparing with `user.ExpiresAt < time.Now().UTC().Add(-(time.Minute*constants.TokenExpiryTimeBufferInMinutes)).Unix()` which also returns `Unix in int64` but `Until` will return `duration` and not the `Unix timestamp`

can you explain this a bit more for which part or operand(left/right) you are suggesting to use `Until`?

Reply:
You can do something like this.
```suggestion
	if user.AccessToken != "" && time.Until(time.Unix(user.ExpiresAt, 0)) <= time.Minute*constants.TokenExpiryTimeBufferInMinutes {
```

Reply:
I think this will not work properly as `time.Until(time.Unix(user.ExpiresAt, 0))` will always be less than `time.Minute*constants.TokenExpiryTimeBufferInMinutes`

pls see below image also, we can disc. on this and if needed we can pick the change in a separate PR merging this for now
![image](https://user-images.githubusercontent.com/72438220/188379979-147e1a53-6a85-45f5-8339-ec93d24d4e34.png)

Reply:
You can do something like this - 
```go
localExpiryTime := time.Unix(user.ExpiresAt, 0).Local()
if time.Until(localExpiryTime) <= time.Minute*constants.TokenExpiryTimeBufferInMinutes {
```